### PR TITLE
feat: added text support to system tray - macos, closes #65

### LINKED ISSUE #65: Implement ability to show text in the system tray/menu bar
Use case: apps like [MeetingBar](https://github.com/leits/MeetingBar) show text instead of a system tray icon.

image

I'd love the ability to do this with Tauri:

- Show text instead of the system tray icon, and be able to update it during runtime.
- Show text _next_ to the system tray icon.

However, I'm not sure if this is even possible in other operating systems. With that in mind, maybe it doesn't fit Tauri's philosophy to add a feature that will only work on macos!

### PR DESCRIPTION
**Pull request by BenJeau**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information

Here's an example of the change, closes #65 

https://user-images.githubusercontent.com/22248828/163900111-fadff001-1de9-42e6-b4d1-1332c0eb2985.mov

As this is my first contribution to tauri, please let me know of anything that needs to be done/changed for this to be accepted to be merged

### CODE REVIEW

File: src/system_tray.rs [hunk 1]
```diff
@@ -105,6 +105,16 @@ impl SystemTray {
...
+  /// Set new tray menu title.
+  ///
+  /// ## Platform-specific
+  ///
+  /// Only works on **macOS**
+  #[cfg(target_os = "macos")]
+  pub fn set_title(&mut self, title: &str) {
+    self.0.set_title(title)
+  }
+
```
Comment (amrbashir) [inline src/system_tray.rs:117]: This should be in `SystemTrayExtMacOS` trait and you should make a builder variant for `SystemTrayBuilderExtMacOS`. 

You can find them in `src/platform/macos.rs`

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks, just a small change needed.

**Comment by patrick91** [comment]
this looks very promising, is there any way I can test this in my app? :)

**Comment by vojty** [comment]
Since I'm waiting for this feature as well I created a new PR https://github.com/tauri-apps/tao/pull/554

The changes are the same as in this PR, I've just moved things to the existing traits as @amrbashir suggested and added support for the builder variant.

**Comment by amrbashir** [comment]
closing in favor of #554. Thanks @BenJeau and @vojty

**Pull request closed without merging**